### PR TITLE
Align logging practices with cluster-api

### PR DESCRIPTION
**Pull request by Skarlso**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3626

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by Skarlso** [comment]
Hmmm, not sure if I'm done yet. CAPA does already some nice logging practices such as using ctx logger in reconcile loops.

But I feel like I'm missing something still... I will look at `V(n)` logs as well later on.

**Comment by Ankitasw** [comment]
This looks good 👍 
@Skarlso V(n) logging is also not consistent, so I think we can handle that as part of this PR.
Wdyt?

**Comment by Skarlso** [comment]
@Ankitasw True. It might warrant a separate PR to handle `V(n)` log numbers. :)

**Comment by Ankitasw** [comment]
Ok, makes sense.
I think [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3152) issue is covering V(n) logging consistency part.

**Comment by sedefsavas** [comment]
There are few other changes done in CAPV like using  "sigs.k8s.io/controller-runtime instead of "sigs.k8s.io/controller-runtime/pkg/log".

https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1602/commits/df945ea3835667c9ce6b35b244e230377f46ee29

Also, does this PR cover the https://cluster-api.sigs.k8s.io/developer/providers/v1.1-to-v1.2.html logging section, there are bunch of changes in main in this doc.

/hold 
until https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3720 is merged.

**Comment by Skarlso** [comment]
Ah, thanks, @sedefsavas!

I shall take a look and incorporate those changes. :)

**Comment by Skarlso** [comment]
I'm not sure what to make of this error. I run `go mod tidy` in in main and in hack/tools and it isn't doing anything.

File: cmd/clusterawsadm/cmd/root.go [hunk 1]
```diff
@@ -102,5 +103,5 @@ func init() {
 }
 
 func initConfig() {
-	logf.SetLogger(logf.NewLogger(logf.WithThreshold(verbosity)))
```
[the remark this answers is not in the document]
Comment (Skarlso) [reply cmd/clusterawsadm/cmd/root.go:105]: actually, this turned out to be `V()`...

File: cmd/clusterawsadm/cmd/root.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ func Execute() {
 
 func init() {
 	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
-	verbosity := flag.CommandLine.Int("v", 2, "Set the log level verbosity.")
+	verbosity = flag.CommandLine.Int("v", 2, "Set the log level verbosity.")
```
Comment (Skarlso) [inline cmd/clusterawsadm/cmd/root.go:96]: I don't think this was ever working before properly. :D 
This was trying to use the package local variable but created a new one instead. So

**Comment by Skarlso** [comment]
```
I0913 09:56:33.117909     373 addons.go:164] "describe output" output=<
	{
	  AddonArn: "arn:aws:eks:eu-central-1:<NOPE>addon/default_managed-cluster-control-plane/vpc-cni/aac19b19-02bc-2d31-5eee-b48ef0cfe5ce",
	  AddonName: "vpc-cni",
	  AddonVersion: "v1.11.0-eksbuild.1",
	  ClusterName: "default_managed-cluster-control-plane",
	  CreatedAt: 2022-09-13 09:33:11.631 +0000 UTC,
	  Health: {
	    Issues: []
	  },
	  ModifiedAt: 2022-09-13 09:33:12.697 +0000 UTC,
	  Status: "ACTIVE",
	  Tags: {
	    kubernetes.io/cluster/managed-cluster: "owned"
	  }
	}
 >
I0913 09:56:33.119742     373 addons.go:95] "Reconcile EKS addons completed successfully"
I0913 09:56:33.119784     373 identity_provider.go:33] "reconciling oidc identity provider"
I0913 09:56:33.119814     373 eks.go:64] "Reconcile EKS control plane completed successfully"
I0913 09:56:33.119846     373 cni.go:45] "Reconciling aws-node DaemonSet in cluster" cluster="default/managed-cluster"
I0913 09:56:33.120667     373 recorder.go:103] "events: Normal" object={Kind:AWSManagedControlPlane Namespace:default Name:managed-cluster-control-plane UID:2fa136ba-67ff-4b1d-9d4f-0e375b6b7fa7 APIVersion:controlplane.cluster.x-k8s.io/v1beta1 ResourceVersion:12502 FieldPath:} reason="SuccessfulReconcileEKSClusterAddons" message="Reconciled addons for EKS Cluster default_managed-cluster-control-plane"
I0913 09:56:33.121198     373 recorder.go:103] "events: Normal" object={Kind:AWSManagedControlPlane Namespace:default Name:managed-cluster-control-plane UID:2fa136ba-67ff-4b1d-9d4f-0e375b6b7fa7 APIVersion:controlplane.cluster.x-k8s.io/v1beta1 ResourceVersion:12502 FieldPath:} reason="SuccessfulReconcileEKSClusterAddons" message="Reconciled addons for EKS Cluster default_managed-cluster-control-plane"
I0913 09:56:33.777756     373 reconcile.go:39] "Reconciling kube-proxy DaemonSet in cluster" cluster="default/managed-cluster"
I0913 09:56:33.903582     373 reconcile.go:34] "Reconciling aws-iam-authenticator configuration" cluster="default/managed-cluster"
I0913 09:56:34.859719     373 reconcile.go:60] "Mapping node IAM role" iam-role="arn:aws:iam::542654844075:role/nodes.cluster-api-provider-aws.sigs.k8s.io" user="system:node:{{EC2PrivateDNSName}}"
I0913 09:56:34.891112     373 reconcile.go:65] "Mapping additional IAM roles and users"
I0913 09:56:34.891191     373 reconcile.go:81] "Reconciled aws-iam-authenticator configuration" cluster-name="default_managed-cluster-control-plane"
```

Logs are looking good. :)

**Comment by Skarlso** [comment]
Oh Gods.... :D

**Pull request merged**